src/plugins/crypto/crypto.c

```diff
@@ -145,6 +146,28 @@ static int elektraCryptoCreateRandomString (Key * errorKey ELEKTRA_UNUSED, char
 }
 
 /**
+ * @brief overwrites the value of the key with zeroes and then releases the Key.
+ * @param key to be overwritten and released
+ */
+static void elektraCryptoSafelyReleaseKey (Key * key)
+{
+	if (key)
+	{
+		// overwrite key content with zeroes
+		ssize_t length = keyGetValueSize (key);
+		kdb_octet_t * newVal = calloc (length, 1);
+		if (newVal)
+		{
+			keySetBinary (key, newVal, length);
```

Comment:
I am afraid in the current implementation there is no guarantee this line will overwrite anything.
(realloc might return a new pointer despite of requesting the same size). We could change keySetRaw to get the guarantee (is trivial and an optimization, too. The code in line 529 should use elektraRealloc btw.).

Or what about simply doing a `memset` of what you get from `keyValue`?